src/azure-cli/azure/cli/command_modules/resource/commands.py

```diff
@@ -335,6 +337,8 @@ def load_command_table(self, _):
                          table_transformer=deployment_validate_table_format, exception_handler=handle_template_based_exception)
         g.custom_command('create', 'deploy_arm_template_at_tenant_scope', supports_no_wait=True, validator=process_deployment_create_namespace,
                          table_transformer=transform_deployment, exception_handler=handle_template_based_exception)
+        g.custom_command('what-if', 'what_if_deploy_arm_template_at_tenant_scope', validator=process_deployment_create_namespace,
```

Comment:
It's better to define a CLIArgumentType for what-if for reuse instead of duplicate multiple places.